design/ironic-debuggability-improvement.md

```diff
@@ -0,0 +1,108 @@
+<!--
+ This work is licensed under a Creative Commons Attribution 3.0
+ Unported License.
+
+ http://creativecommons.org/licenses/by/3.0/legalcode
+-->
+
+# ironic-debuggability-improvement
+
+## Status
+
+provisional
+
+## Summary
+
+Ironic and Ironic-inspector logs are currently hidden inside their respective
+containers and never shown to the end user by default.
+This document describes proposed changes in Ironic and Ironic-inspector
+logs handling to make debugging easier for the end user.
+
+## Motivation
+
+We need to make Ironic logs accessible to the end user.
+
+### Goals
+
+1. Make it easier to understand what happened when a deployment fails.
+2. Avoid old logs pile up and take all available space.
+
+### Non-Goals
+
+1. Modify Ironic logging events or log level.
+
+## Proposal
+
+The ironic-inspector-image includes a new script, ironic-inspection-log-watch,
+that can act as an entry point for a container to dump host inspection logs.
+
+The script should watch for Ironic host inspection log files to appear in
+"/shared/logs/ironic-inspector/ramdisk", wait for them to stop changing
+(to ensure the entire log is dumped), decompress them, print their contents
+with each line prefixed by the base file name, and then remove the file.
+
+The ironic-image includes a new script, ironic-provisioning-log-watch,
+that can act as an entry point for a container to dump host provisioning logs.
```

Comment:
We're talking here about adding a script to both the ironic-image and the ironic-inspector-image images,
with the exception of the directory being checked I'd imagine the scripts will be similar (identical?),
could we avoid some duplication and maintenance by only adding this to the ironic-image image and have
it monitor both directories or parameterize the directory?

Reply:
or in the ironic-client image

Reply:
> "It's a shame that ironic doesn't offer the option to include the name in the filename format, as that would make the log output a bit more easily searchable (particularly in the case where the metal3 pod gets rescheduled so all the UUIDs change?)"

Just noting that the "Instance UUID" field is part of the filename for the deploy logs (not inspector), which is the bmh uid.

Reply:
I would agree with adding a unified log monitoring script into ironic-client image. We are not taking this path with the new Ironic changes though. Otherwise I would let each image to take care of their own logs to avoid creating new dependencies.

Reply:
ok